### DISCUSSION TITLE
Some help with Layouts and /home needed after app re-imported

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#2031] [Help] [answered]
Hello Andrew @aptkingston 

Could you please help me with the following:

Finally I decided to re-import my app (created in old versions), and also decided to re-create my ex layouts by copy/paste containers in new layouts (just do not want to deal with overflow issues in my old existing layouts after app reimported). 
And also I want to create more new ones on the basis of style of exiting old layouts.

In re-imported app, I created new layout and copied/pasted containers from existing old layout.
New layout Width was set to Max, 
The problem is:  if I add `align-items: center;` in Custom CSS, then the width of whole layout decreases. 
Adding `width: 100%; min-width: 100%; `does not solve the problem.

It shown on the following screens:

**Existing old layout** (nav links container not yet fixed after app re-imported, I'll fix it later) :

![01_exisitng](https://user-images.githubusercontent.com/53024832/126333678-434a57d7-2ad2-47b9-812f-1ce356dd1f1a.png)

**Newly created Layout with containers I copied from existing old layout:**

![02-new](https://user-images.githubusercontent.com/53024832/126333904-430afc91-7698-4afd-ac9a-6306252172da.png)

**Trying to move Screen Slot to center** (`align-items: center;` in Custom CSS of the layout) - the padding appears and whole layout width decreases:

![03-new](https://user-images.githubusercontent.com/53024832/126334175-5e335a8f-ca8b-4d93-bad9-8c256df669cd.png)
 

This Screen Slot has width: 1200px, min-width 1200px and should be at center (horizontal) of the layout while Header and Footer are 100% page width.

Another issue I faced: I have a dashboard screen in my old app at "/" (home). It is gone after app re-imported. Instead of my complex dashboard screen slot I got empty page with Welcome to Budibase message.

Thank you.

**Comment by GuilhermeHirsch** [comment]
Helo, @sergey-dme! Regarding the first situation described by you. This margin can be corrected by setting a negative margin value in "Screenslot", as in the following image:

![Sem título](https://user-images.githubusercontent.com/33108596/126336504-f194e90e-18b2-41a8-b8ac-f3fe710a04a2.jpg)

**Reply by sergey-dme** [reply]
Hi @GuilhermeHirsch 
Unfortunately it does not solve the problem. If I add neg margin in Screen Slot CSS as you suggested, my Header and Footer squeezes even more:

![screen_slot_maring_neg](https://user-images.githubusercontent.com/53024832/126342319-b7ada35b-d487-4a4b-8467-702bac638422.png)
![screen_slot_maring_neg_](https://user-images.githubusercontent.com/53024832/126342342-e9bd08d9-1a17-423a-b994-878e081fd99f.png)
P.S. Adding neg margin in Header and Footer CSS also does not help.

**What I am trying to re-create:** My Screen Slot has width: 1200px, min-width 1200px and should be at center (horizontal) of the layout, while Header and Footer are 100% page width as it in my old existing layouts. Unbale to recreate it in new layout... 
The only way for now, to have my Footer/Headers 100% page with is removing` align-items: center` from Layout CSS. But in this case Screen Slot (1200xp width) stays at the Left side of the layout, not in the center (shown in my first message)

**Comment by sergey-dme** [comment]
Currenly it can be solved by putting all my 3 parts (Header Cont, Screen Slot, Footer Cont) in a new Container with the fowling CSS:

`height: 100%;`
` width: 100%;`
`overflow-y: auto;`
` overflow-x: hidden;`
`position: relative;`
`align-items: center;`

Btw, thanks for the new "Max" option in Layouts!

But can't figure out why my dashboard at /home is gone and replaced by empty page with Welcome headline after app re-import

**Comment by aptkingston** [comment, marked as answer]
@sergey-dme I've tried to replicate this with a new app using the latest release, but it works perfectly for me. Even on ultrawide resolutions, you can see that the screen slot fills the entire width of the layout.
![image](https://user-images.githubusercontent.com/9075550/126460557-3de6c4bc-d01d-449b-ae21-1e4eb2752f24.png)
This must be a problem due to all of your custom CSS rules, because with no custom styles applied it works as intended. Please try removing all custom rules from the layout and make sure the screen slot is not nested inside any of your containers and remove all custom CSS from the screen slot too and see if that helps.

**Reply by sergey-dme** [reply]
@aptkingston Thank you. I removed all. 

Can you please try to set Screen slot fixed width like 1200px for example, and header is 100% page width. 

The problem was that 1200px Screen slot was at the left side, not at the page center. And CSS `align-items: center` didn't fix it - header is squeezed after it. The solution I found is placing custom header (which is 100% width) and Screen Slot 1200px in a new same container - it works.

**Reply by aptkingston** [reply]
Ah, I see. The easiest way to do that is simply add 2 custom rules to the screen slot: `width` and `align-self`. Leave the layout width setting at `Max`. Here's a screenshot:
![image](https://user-images.githubusercontent.com/9075550/126464289-fc74083d-0fe3-43cf-a1d6-81147e89eff4.png)

**Reply by sergey-dme** [reply]
@aptkingston Cool! Thanks a lot!
